target/linux/ath79/image/common-yuncore.mk

```diff
@@ -0,0 +1,11 @@
+define Build/yuncore-tftp-header-qca953x
+	( \
+		echo -n -e "YUNCOREsetenv bootcmd \"bootm 0x9f050000 || bootm 0x9fe80000\" && " \
+			"setenv resetcmd \"erase 0x9f050000 +0xfa0000 && cp.b 0x800600c0 0x9f050000 0xfa0000\" && " \
+			"setenv bootargs \"console=ttyS0,115200 rootfstype=squashfs,jffs2 noinitrd\" && " \
```

Comment:
Resetcmd is used by custom uboot of the device.

Reply:
I removed the bootargs , and kept the one in DTS

Reply:
No. To clarify:

The resetcmd saved in the U-Boot environment variable is executed when there is no resetcmd defined at the beginning of the image. By default there is no resetcmd defined. As we define a resetcmd in the firmware image, no resetcmd should be set in the environment variable.

**Furthermore this script is out-of-scope for this PR as I've opened up a separate PR for this.
Stealing my code to this PR is very disrespectful.**

Reply:
@CodeFetch Your code just is not working. Please do keep polite and avoid bold  and nervous comments, this is totally not usefull

Removing the resetcmd here makes it that there is no resetcmd nowhere. The resetcmd must be initiated somewhere. So if not here, then where ?

Reply:
I sourced a yuncire 830, which has UART

This is the error I get :

```
Using eth0 device
TFTP from server 192.168.0.141; our IP address is 192.168.0.144
Filename 'upgrade.bin'.
Load address: 0x80060000
Loading: #################################################################
         #################################################################
         #################################################################
         #################################################################
         #################################################################
         #################################################################
         #################################################################
         #################################################################
         #################################################################
         #################################################################
         #################################################################
         #################################################################
         #################################################################
         #############
done
Bytes transferred = 4391347 (4301b3 hex)
firmware resetcmd string: YUNCOREsetenv bootcmd "bootm 0x9f050000 || bootm 0x9fe80000" &&  setenv resetcmd "erase 0x9f050000 +0xfa0000 
Please wait......
syntax error

NOTIFICATION: firmware resetcmd FINISHED
Rebooting......

```

Reply:
@grosjo Please upload the resulting firmware image so I can have a look.

Reply:
And please try the version from my PR without your changes...

Reply:
I've found your problem... Look at the dd-line. The command exceeds 192 characters with your addition of the resetcmd (221). Thus it gets cut.

https://github.com/openwrt/openwrt/pull/2362/files#diff-f98caa543f5e6dc60bf413ccf8117b69R6

> Removing the resetcmd here makes it that there is no resetcmd nowhere. The resetcmd must be initiated somewhere. So if not here, then where ?

The resetcmd is temporarily defined by the command string at the beginning of the image.

YunCore's U-Boot TFTP  recovery works like this:
1. Load image via TFTP
2. Check whether there is the magic string "YUNCORE" at the beginning of the image
	If yes:
		Execute following command up to a NULL byte
	If no:
		Check whether resetcmd environment variable is set
		If yes:
			Execute command set in resetcmd environment variable
		If no:
			Print error message

Thus you should not set the resetcmd environment variable, because once you've done that the router will accept any image via TFTP and execute the command set in the variable every time...

Reply:
No, you are absolutely wrong

I removed the so called 'keyword'  YUNCORE and then this solve this first error.
However, there is a second error (see below)

```
TFTP from server 192.168.0.141; our IP address is 192.168.0.144
Filename 'upgrade.bin'.
Load address: 0x80060000
Loading: #################################################################
         #################################################################
         #################################################################
         #################################################################
         #################################################################
         #################################################################
         #################################################################
         #################################################################
         #################################################################
         #################################################################
         #################################################################
         #################################################################
         #################################################################
         #############
done
Bytes transferred = 4391347 (4301b3 hex)
firmware resetcmd string: setenv bootcmd "bootm 0x9f050000 || bootm 0x9fe80000" &&  saveenv && erase 0x9f050000 +0xfa0000 && cp.b 0x80 
no reset button command in firmware
checking the reset button cmd in env
env resetcmd string: erase 0x9f050000 +0xFA0000&&cp.b 0x80060000 0x9f050000 $filesize 
Please wait......
Erasing flash... 
First 0x5 last 0xfe sector size 0x10000
 254
Erased 250 sectors
Copy to Flash... write addr: 9f050000
done

NOTIFICATION: env resetcmd FINISHED
Rebooting......

eth1 up
eth0, eth1
Setting 0x181162c0 to 0x409a100
Hit any key to stop autoboot:  0 
## Booting image at 9fe80000 ...
Bad Magic Number
## Booting image at 9f050000 ...
Bad Magic Number

```

Reply:
It is now working fine